src/elbo.jl

```diff
@@ -0,0 +1,247 @@
+ScalarLikelihood = Union{
+    BernoulliLikelihood,
+    PoissonLikelihood,
+    GaussianLikelihood,
+    ExponentialLikelihood,
+    GammaLikelihood
+}
+
+
+abstract type ExpectationMethod end
+struct Default <: ExpectationMethod end
+struct Analytic <: ExpectationMethod end
+
+struct Quadrature <: ExpectationMethod
+    n_points::Int
+end
+Quadrature() = Quadrature(20)
+
+struct MonteCarlo <: ExpectationMethod
+    n_samples::Int
+end
+MonteCarlo() = MonteCarlo(20)
+
+"""
+    elbo(fx::FiniteGP, y::AbstractVector{<:Real}, fz::FiniteGP, q::AbstractMvNormal; n_data=length(y), method=Default())
+
+Compute the Evidence Lower BOund from [1] for the process `fx.f` where `y` are
+observations of `fx`, pseudo-inputs are given by `z = fz.x` and `q(u)` is a
+variational distribution over inducing points `u = f(z)`.
+
+`method` selects which method is used to calculate the expected loglikelihood in
+the ELBO. The options are: `Default()`, `Analytic()`, `Quadrature()` and
+`MonteCarlo()`. For likelihoods with an analytic solution, `Default()` uses this
+exact solution. If there is no such solution, `Default()` either uses
+`Quadrature()` or `MonteCarlo()`, depending on the likelihood.
+
+N.B. the observation noise `fx.Σy` is assumed to be homoscedastic and
+uncorrelated - i.e. only `fx.Σy[1]` is used.
+
+[1] - Hensman, James, Alexander Matthews, and Zoubin Ghahramani. "Scalable
+variational Gaussian process classification." Artificial Intelligence and
+Statistics. PMLR, 2015.
+"""
+function AbstractGPs.elbo(
+    fx::FiniteGP{<:AbstractGP, <:AbstractVector, <:Diagonal{<:Real, <:Fill}},
+    y::AbstractVector{<:Real},
+    fz::FiniteGP,
+    q::AbstractMvNormal;
+    n_data=length(y),
+    method=Default()
+)
+    return _elbo(method, fx, y, fz, q, GaussianLikelihood(fx.Σy[1]), n_data)
+end
+
+function AbstractGPs.elbo(::FiniteGP, ::AbstractVector, ::FiniteGP, ::AbstractMvNormal; kwargs...)
+    return error("The observation noise fx.Σy must be homoscedastic.\n To avoid this error, construct fx using: f = GP(kernel); fx = f(x, σ²)")
+end
+
+"""
+    elbo(lfx::LatentFiniteGP, y::AbstractVector, fz::FiniteGP, q::AbstractMvNormal; n_data=length(y), method=Default())
+
+Compute the ELBO for a LatentGP with a possibly non-conjugate likelihood.
+"""
+function AbstractGPs.elbo(
+    lfx::LatentFiniteGP,
+    y::AbstractVector,
+    fz::FiniteGP,
+    q::AbstractMvNormal;
+    n_data=length(y),
+    method=Default()
+)
+    return _elbo(method, lfx.fx, y, fz, q, lfx.lik, n_data)
+end
+
+# Compute the common elements of the ELBO
+function _elbo(
+    method::ExpectationMethod,
+    fx::FiniteGP,
+    y::AbstractVector,
+    fz::FiniteGP,
+    q::AbstractMvNormal,
+    lik::ScalarLikelihood,
+    n_data::Integer
+)
+    post = approx_posterior(SVGP(), fz, q)
+    f_mean, f_var = mean_and_var(post, fx.x)
+    variational_exp = expected_loglik(method, y, f_mean, f_var, lik)
+
+    kl_term = StatsBase.kldivergence(q, fz)
```

Comment:
```suggestion
    kl_term = kldivergence(q, fz)
```
should work just the same